Проект 1

minchik opened the pull request.

minchik committed:
1 Project

mamahtehok commented on DNA/Cell.m:
```diff
@@ -0,0 +1,68 @@
+            
+            curr_cell_type = arc4random() % 4;
+            
+            switch (curr_cell_type) {
```
Генерацию значения лучше бы вынести в отдельный метод, чтоб не повторять  его несколько раз.

mamahtehok commented on DNA/main.m:
```diff
@@ -0,0 +1,86 @@
+
+
+@interface Cell (Mutator)
+    
```
Категорию лучше разместить в отдельном файле

mamahtehok commented on DNA/main.m:
```diff
@@ -0,0 +1,86 @@
+                new_cell_name = @"C";
+                break;
+        }
+        [dna replaceObjectAtIndex:[cell_to_modify intValue] withObject:new_cell_name];
```
Нет проверки на совпадение значения гена.

minchik committed:
Final

Перенес категорию в отдельый файл.
Вынес случайную инициализацию гена в отдельный метод.
Добавил проверку на совпадение старого значения гена с новым

mamahtehok commented on Cell+Mutator.m:
```diff
@@ -0,0 +1,35 @@
+    NSNumber *cell_to_modify = 0;
+    NSMutableSet * modifiedCells = [[NSMutableSet alloc] init];
+    NSString *curr_cell_name = [[NSString alloc]init];
+    while (f!=0) {
```
В данном случае у вас получается изменение f ячеек, а должно менять f% массива. При размерности массива в 100 это эквивалентно  но если размерность массива поменять то будет уже не то. Переделайте чтоб меняло f% , т.к. данное решение не соответствует условиям задачи.

minchik committed:
Final

Добавил правильный расчет процентов (в случае если количество генов в
ячейка будет меняться со временем)
